pkg/applicationserver/io/packages/loragls/v3/api/client.go

```diff
@@ -0,0 +1,133 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+
+	urlutil "go.thethings.network/lorawan-stack/v3/pkg/util/url"
+	"go.thethings.network/lorawan-stack/v3/pkg/version"
+)
+
+// Option is an option for the API client.
+type Option interface {
+	apply(*Client)
+}
+
+// OptionFunc is an Option implemented as a function.
+type OptionFunc func(*Client)
+
+func (f OptionFunc) apply(c *Client) { f(c) }
+
+// Client is an API client for the LoRa Cloud Device Management v1 service.
+type Client struct {
+	token   string
+	baseURL *url.URL
+	cl      *http.Client
+}
+
+const (
+	contentType      = "application/json"
+	defaultServerURL = "https://gls.loracloud.com"
+	basePath         = "/api/v3"
```

Comment:
Can this be made configurable in case out upstream friends decide to change this overnight?

Reply:
If they change this overnight, most likely they would change the request / response format, case in which the package is unusable anyway. But the server URL _is_ configurable.